Runtime/PostProcessing/CinemachinePostProcessing.cs

```diff
@@ -276,21 +276,30 @@ static void ApplyPostFX(CinemachineBrain brain)
 
         static PostProcessLayer GetPPLayer(CinemachineBrain brain)
         {
-            PostProcessLayer layer = null;
-            if (mBrainToLayer.TryGetValue(brain, out layer))
+            bool found = mBrainToLayer.TryGetValue(brain, out PostProcessLayer layer);
+            if (found)
             {
 #if UNITY_EDITOR
                 // Maybe they added it since we last checked
                 if (layer != null || Application.isPlaying)
 #endif
                 return layer;
             }
+#if UNITY_2019_2_OR_NEWER
+            brain.TryGetComponent(out layer);
+#else
             layer = brain.GetComponent<PostProcessLayer>();
-            mBrainToLayer[brain] = layer;
-            if (layer != null)
-                brain.m_CameraCutEvent.AddListener(OnCameraCut);
-            else
+#endif
+            if (found)
```

Comment:
if found is true, then we returned at line 286.
Except, if UNITY_EDITOR is true, and (layer != null || Application.isPlaying) is false.

Is this intentional?

Reply:
Yes.  If not playing, we allow for the possibility of the user adding a PP layer at any time, because they are still editing the scene.  We don't bother with this check if playing, to save on processing time where it matters.

Reply:
I agree that this code is very confusing... and not quite correct.  I will rewrite